account-kit/react/src/components/auth/hooks/useOAuthVerify.ts

```diff
@@ -0,0 +1,58 @@
+"use client";
+import { useCallback } from "react";
+import { useAuthenticate } from "../../../hooks/useAuthenticate.js";
+import { useAuthContext } from "../context.js";
+import type { AuthType } from "../types.js";
+
+export type UseOAuthVerifyReturnType = {
+  authenticate: () => void;
+  isPending: boolean;
+};
+
+export const useOAuthVerify = ({
+  type,
+  ...config
+}: Extract<AuthType, { type: "social" }>): UseOAuthVerifyReturnType => {
+  const { setAuthStep } = useAuthContext();
+
+  const { authenticate: authenticate_, isPending } = useAuthenticate({
+    onMutate: () => {
+      setAuthStep({
+        type: "oauth_completing",
+        provider:
+          config.authProviderId === "auth0"
+            ? config.auth0Connection!
+            : config.authProviderId,
+        ...(config.authProviderId === "auth0" &&
+          "logoUrl" in config && { logoUrl: config.logoUrl }),
+      });
+    },
+    onError: (err) => {
+      setAuthStep({
+        type: "oauth_completing",
+        provider:
+          config.authProviderId === "auth0"
+            ? config.auth0Connection!
+            : config.authProviderId,
+        error: err,
+        ...(config.authProviderId === "auth0" &&
+          "logoUrl" in config && { logoUrl: config.logoUrl }),
```

Comment:
nit: for this you can get the `authStep` returned from `useAuthContext` and just spread it here with the error added to it

Reply:
for `onError` when I `setAuthStep` to `{...authStep, error: err}`, there is an error because the other `AuthStep` definitions don't have an `error` field